cypress/tests/ui/flowchart/flowchart.cy.js

```diff
@@ -1 +1,166 @@
+  it('verifies that users can open the metadata panel for nodes. #TC-24', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--task').first().click();
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+  });
+
+  it('verifies that users can open the code block in the metadata panel. #TC-25', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata-code--visible').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--task').first().click();
+    cy.get('[data-test="pipeline-toggle-input-code"]').check({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata-code--visible').should('exist');
+    cy.get('.pipeline-metadata-code__title').should('have.text', 'Code block');
+  });
+
+  it('verifies that users can open the metadata panel for parameters. #TC-26', () => {
+    const nodeToToggleText = 'Parameters';
+
+    // Alias
+    cy.get(`.pipeline-nodelist__row__checkbox[name=${nodeToToggleText}]`).as(
+      'nodeToToggle'
+    );
+
+    // Assert before action
+    cy.get('@nodeToToggle').should('not.be.checked');
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.get('@nodeToToggle').check({ force: true });
+    cy.get('.pipeline-node--parameters').first().click({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+  });
+
+  it('verifies that users can open the metadata panel for datasets. #TC-27', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--data').first().click();
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+  });
+
+  it('verifies that users can open and preview the dataset table in the metadata panel for datasets. #TC-28', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata-modal').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--data').first().click();
+    cy.get('.pipeline-metadata__link').click();
+
+    // Assert after action
+    cy.get('.pipeline-metadata-modal').should('exist');
+    cy.get('.pipeline-metadata-modal__preview').should('exist');
+    cy.get('.preview-table').should('exist');
+  });
+
+  it('verifies that users can open the metadata panel showing the Plotly/Matplotlib preview. #TC-29', () => {
+    const nodeToClickText = 'Price Histogram';
+
+    // Assert before action
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+    cy.get('.plot-container.plotly').should('not.exist');
+
+    // Action
+    cy.contains('text', nodeToClickText).click({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+    cy.get('.plot-container.plotly').should('exist');
```

Comment:
Yup, like in line 104 ... if we can have an extra assertion in above 2

Reply:
got it :)